pkg/model/awsmodel/autoscalinggroup.go

```diff
@@ -397,7 +397,8 @@ func (b *AutoscalingGroupModelBuilder) buildAutoScalingGroupTask(c *fi.ModelBuil
 			"GroupTotalInstances",
 		},
 
-		InstanceProtection: fi.Bool(false),
+		InstanceProtection:  fi.Bool(false),
+		MaxInstanceLifetime: ig.Spec.MaxInstanceLifetime,
```

Comment:
Here you may want to set `fi.Int64(0)` if `ig.Spec.MaxInstanceLifetime` is `nil`. In order to unset the value with AWS, you have to pass `0`. Not setting the value will mean the value should be unchanged.

This is a similar issue to https://github.com/kubernetes/kops/issues/13849 (fixed by https://github.com/kubernetes/kops/pull/13852)